src/main/java/org/jenkinsci/plugins/displayurlapi/actions/AbstractDisplayAction.java

```diff
@@ -45,27 +44,21 @@ public final Object doRedirect(StaplerRequest req, StaplerResponse rsp) throws I
 
     DisplayURLProvider lookupProvider(StaplerRequest req) {
         final String providerName = req.getParameter("provider");
-        if(providerName != null && !providerName.isEmpty()) {
-            Iterable<DisplayURLProvider> providers = DisplayURLProvider.all();
-            Iterable<DisplayURLProvider> filtered = Iterables.filter(providers, new Predicate<DisplayURLProvider>() {
-                @Override
-                public boolean apply(@Nullable DisplayURLProvider displayURLProvider) {
-                    if(displayURLProvider == null) {
-                        return false;
-                    }
-
-                    return displayURLProvider.getName().equals(providerName);
-                }
-            });
-
-            DisplayURLProvider provider = Iterables.getFirst(filtered, null);
+        if(StringUtils.isNotEmpty(providerName)) {
+            ExtensionList<DisplayURLProvider> providers = DisplayURLProvider.all();
+            DisplayURLProvider provider = providers.stream()
+                .filter(displayURLProvider -> providerName.equals(displayURLProvider.getName()))
```

Comment:
Handling `displayURLProvider == null` is not required anymore?

Reply:
Nicely spotted 😄

Reply:
Fixed 👍